api/status.go

```diff
@@ -41,6 +42,48 @@ func getStatusesHttp(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.StringInterfaceToJson(statusMap)))
 }
 
+func getTeamStatusesHttp(c *Context, w http.ResponseWriter, r *http.Request) {
+	statusMap := map[string]interface{}{}
+
+	if result := <-Srv.Store.Status().GetAllFromTeam(c.TeamId); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else {
+		statuses := result.Data.([]*model.Status)
+		for _, s := range statuses {
+			statusMap[s.UserId] = s.LastActivityAt
+		}
+	}
+
+	if result := <-Srv.Store.User().GetProfiles(c.TeamId); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else {
+		profiles := result.Data.(map[string]*model.User)
+
+		for k, p := range profiles {
+			options := utils.Cfg.GetSanitizeOptions()
```

Comment:
This should be moved over to the new sanitize function after https://github.com/mattermost/platform/pull/3930 is merged.